apps/github-cascading-app/src/cascading/cascading.ts

```diff
@@ -0,0 +1,392 @@
+   * @param targetBranch name of the target branch (base of the pull request)
+   * @param baseVersion
+   * @param targetVersion
+   */
+  protected determineCascadingBranchName(baseVersion: string, targetVersion: string) {
+    return `cascading/${baseVersion}-${targetVersion}`;
+  }
+
+  /**
+   * Find the pull request open with for the given branches
+   *
+   * @param baseBranch name of the base branch (head of the pull request)
+   * @param targetBranch name of the target branch (base of the pull request)
+   */
+  protected async findOpenPullRequest(baseBranch: string, targetBranch: string) {
+    const pRs = await this.getPullRequests(baseBranch, targetBranch);
+    const openPr = pRs.find((pr) => pr.isOpen);
+    return openPr;
+  }
+
+
+  /**
+   * Generate the pull request body part added in case of cascading re-evaluation request
+   *
+   * @param baseBranch Base branch of the pull request
+   * @returns
+   */
+  protected createMessageNewUpdate(baseBranch: string) {
+    return `
+---
+${MESSAGE_NEW_UPDATE} ${baseBranch}
+---`;
+  }
+
+  /**
+   * Add update message to the pull request body to request re-evaluation of the base branch
+   *
+   * @param cascadingBranch name of the cascading branch (head of the pull request)
+   * @param currentBranch name of the base branch
+   * @param targetBranch name of the target branch (base of the pull request)
+   */
+  protected async addTriggerToPullRequest(cascadingBranch: string, currentBranch: string, targetBranch: string) {
+    this.logger.debug(`Run trigger to cascading PR from ${cascadingBranch}`);
+    const openPr = await this.findOpenPullRequest(cascadingBranch, targetBranch);
+
+    if (!openPr) {
+      try {
+        await this.deleteRemoteBranch(cascadingBranch);
+        await this.createBranch(cascadingBranch);
+        await this.push(cascadingBranch);
+      } catch (error) {
+        this.logger.warn(`Failed to renew the cascading branch ${cascadingBranch}`);
+        this.logger.debug(JSON.stringify(error, null, 2));
+      }
+      await this.createPullRequestWithMessage(cascadingBranch, currentBranch, targetBranch, true);
+    } else {
+      let message = openPr.body || '';
+      const triggerMessage = this.createMessageNewUpdate(currentBranch);
+      const triggerUncheckMessage = triggerMessage.replace('[x]', '[ ]');
+      if (message.includes(triggerUncheckMessage)) {
+        message.replace(triggerUncheckMessage, triggerMessage);
+      } else if (!message.includes(triggerMessage)) {
+        const footerIndex = openPr.body?.indexOf(MESSAGE_FOOTER);
+        if (footerIndex === undefined || footerIndex < 0) {
+          message += this.createMessageNewUpdate(currentBranch) + MESSAGE_FOOTER;
+        } else {
+          message = message.slice(0, footerIndex) + this.createMessageNewUpdate(currentBranch) + message.slice(footerIndex);
+        }
+      }
+
+      await this.updatePullRequestMessage(openPr.id, message);
```

Comment:
```suggestion
      return this.updatePullRequestMessage(openPr.id, message);
```
For the same reason as before